flow/designs/asap7/riscv32i/rules-base.json

```diff
@@ -1,18 +1,18 @@
 {
     "synth__design__instance__area__stdcell": {
-        "value": 437.4,
+        "value": 0,
         "compare": "<="
     },
```

Comment:
Something went wrong here - the area can't be 0

Reply:
Will check, the files were generated by changing the 2 variables followed by
```
$ make
$ make metadata
$ make update_ok
```

Reply:
The data stored in the `synth_stat.txt` is `Chip area for module '\riscv': 0.043740`

Which is accurately captured in the `metadata-base-ok.json` 

But the `genRuleFile.py` seems to be padding this by 15% and rounding it, which from the math, would result in 0

First, is the `synth__design__instance__area__stdcell` metric supposed to capture only the  `risc_v` module or the complete `riscv_top` module?

```
=== design hierarchy ===

   riscv_top                         1
     $paramod\mux2\WIDTH=s32'00000000000000000000000000100000      1
     ROM                             1
     dmem                            1
     riscv                           1
       controller                    1
         $paramod\mux2\WIDTH=s32'00000000000000000000000000000001      1
         $paramod\mux2\WIDTH=s32'00000000000000000000000000000100      1
         aludec                      1
         maindec                     1
       datapath                      1
         $paramod\flopr\WIDTH=s32'00000000000000000000000000100000      1
         $paramod\mux2\WIDTH=s32'00000000000000000000000000000101      1
         $paramod\mux2\WIDTH=s32'00000000000000000000000000100000      9
         $paramod\mux3\WIDTH=s32'00000000000000000000000000100000      2
         adder                       2
         alu                         1
         magcompare32                1
           magcompare2b             16
           magcompare2c             15
         regfile                     1
         shifter                     1
         signext                     1
           $paramod\mux5\WIDTH=s32'00000000000000000000000000100000      1
```
this could be achieved by changing the expression that is used to extract the data from the input file.

Or should the way the metric is stored by  `genRuleFile.py` be changed to account for the error in math?

[synth_stat.txt](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/files/11014977/synth_stat.txt)

Reply:
@yathAg 
Do `make clean_metadata` and run `make metadata` and check.

Reply:
@vijayank88 for `synth__design__instance__area__stdcell` according to the `genMetrics.py`, the data is copied from the `synth_stat.txt` file and stored in the metadata file. 

```
    extractTagFromFile('synth__design__instance__area__stdcell',
                       metrics_dict,
                       'Chip area for module.*: +(\S+)',
                       rptPath + '/synth_stat.txt')
```


That seems to be perfectly fine, and copying the value from `Chip area for module '\riscv': 0.043740` 


The error is produced when `genRuleFile.py` uses this data to update the `rules-base.json ` file.

What I want to understand is 
1) is the math used in `genRuleFile.py` wrong and could be updated?
2) is the area of the wrong module being copied from `synth_stat.txt` ?

Reply:
In my opinion 
```
extractTagFromFile('synth__design__instance__area__stdcell',
                       metrics_dict,
                       'Chip area for module.*: +(\S+)',
                       rptPath + '/synth_stat.txt')
```

Should be changed to 
extractTagFromFile('synth__design__instance__area__stdcell',
                       metrics_dict,
                       'Chip area for top module.*: +(\S+)',
                       rptPath + '/synth_stat.txt')

Reply:
@yathAg 
Check this [comment](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/pull/836#issuecomment-1442910760). We already discussed.

Reply:
@yathAg you can create new PR with above changes to `genRuleFile.py`. We will check how CI goes through.
Also mention about round off issue during the PR creation with above message

Reply:
I have seemed to figure out the issue @vijayank88 @maliberty 

When `export SYNTH_HIERARCHICAL = 1` is defined in the config.mk do the hierarchical synthesis, the format of the generated synth_stat is different.

[synth_stat_flat.txt](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/files/11017803/synth_stat_flat.txt)
[synth_stat_hier.txt](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/files/11017804/synth_stat_hier.txt)

The current genMetrics.py file only accounts for the flat synthesis and hence reports the wrong area.

I have modified the file to 

```
hierarchicalSynth = os.environ.get('SYNTH_HIERARCHICAL')

    if hierarchicalSynth == '1': 
        extractTagFromFile('synth__design__instance__area__stdcell',
                            metrics_dict,
                            'Chip area for top module.*: +(\S+)',
                            rptPath + '/synth_stat.txt')
    else:
        extractTagFromFile('synth__design__instance__area__stdcell',
                            metrics_dict,
                            'Chip area for module.*: +(\S+)',
                            rptPath + '/synth_stat.txt')
```

And this accounts for both designs.